system/jlib/jlog.cpp

```diff
@@ -2464,15 +2467,33 @@ void setupContainerizedLogMsgHandler()
         unsigned postMortemLines = logConfig->getPropInt(capturePostMortemAtt, 0);
         if (postMortemLines)
         {
+            PostMortemLogMsgHandler *previousPostMortemHandler{nullptr};
+            if (thePostMortemHandler)
+            {
+                previousPostMortemHandler = thePostMortemHandler;
+                queryLogMsgManager()->removeMonitor(thePostMortemHandler);
+            }
+
             // augment postmortem files with <pid> to avoid clashes where multiple processes are running within
             // same process space, e.g. hthor processes running in same k8s container
             unsigned pid = GetCurrentProcessId();
             VStringBuffer portMortemFileBase("/tmp/postmortem.%u.log", pid);
 
             thePostMortemHandler = new PostMortemLogMsgHandler(portMortemFileBase, postMortemLines, MSGFIELD_STANDARD);
             queryLogMsgManager()->addMonitor(thePostMortemHandler, getCategoryLogMsgFilter(MSGAUD_all, MSGCLS_all, TopDetail));
+
+            delete previousPostMortemHandler;
         }
     }
+
+    if (isContainerized())
+    {
+        auto updateConfigFunc = [](const IPropertyTree *oldComponentConfiguration, const IPropertyTree *oldGlobalConfiguration)
+        {
+            setupContainerizedLogMsgHandler();
+        };
+        configUpdateHook.installOnce(updateConfigFunc, false);
```

Comment:
should ensure called when disabled or logConfig missing (see top of setupContainerizedLogMsgHandler),
because could e.g. be disabled at outset, then enabled